_posts/2016-12-06-future.md

```diff
@@ -0,0 +1,444 @@
+
+> In the screenshot above, the top-left pane is the input code and the bottom-right pane is the output code after the babel plugin is run. In this case it's preserving the whitespace of the input code when possible.
+
+By passing in [Recast](https://github.com/benjamn/recast) in the options, Babel can power the future of codemods.
+
+`.babelrc`
+
+```js
+{
+  "parserOpts": {
+    "parser": "recast"
+  },
+  "generatorOpts": {
+    "generator": "recast"
+  }
+}
+```
+
+Run the relevant Babel transform(s) on the source code and overwrite it:
+
+```sh
+babel src -d src 
+```
+
+This feature was just made possible so we're looking forward to making it easier to use and seeing the transformations it can enable. Check out the [6.16.0 blog post](https://babeljs.io/blog/2016/09/28/6.16.0#new-feature) for more info!
+
+Other relevant projects: [JSCodeshift](https://github.com/facebook/jscodeshift), [Recast](https://github.com/benjamn/recast), [Lebab](https://github.com/lebab/lebab).
+
+Relevant Issues:
+
+- [Rewriting Lebab as a Babel transformer](https://github.com/lebab/lebab/issues/138)
+- [Babel integration with jscodeshift](https://github.com/facebook/jscodeshift/issues/168)
+
+### Code Coverage / Instrumentation
+
+We want to support tools like [nyc](https://github.com/istanbuljs/nyc) and [babel-plugin-instanbul](https://github.com/istanbuljs/babel-plugin-istanbul).
+
+### Plugin Ecosystem
+
+Thanks to our vibrant community, new plugins are constantly being created: whether it be a new way to write your [css in jsx](https://github.com/zeit/styled-jsx) or [emojification](https://twitter.com/babeljs/status/604367341380616192).
+
+> Currently there are > 1200 [babel-plugins on npm](https://www.npmjs.com/search?q=babel-plugin-). 
+
+We've had some interesting discussions on how we can grow and support the plugin ecosystem. We could try to watch all the repos but that is obviously overwhelming.
+
+It might be interesting to create some bots to automate a few tasks: create specific Babel plugins/ESLint rules for babel-plugins, write codemods to update API changes, and integrate plugins into our smoke test.
+
+### Documentation/Website
+
+Docs contributions have definitely been lacking in the [last year](https://github.com/babel/babel.github.io/graphs/contributors?from=2015-11-29&to=2016-11-29&type=c).
+
+However just recently we've done a lot of awesome stuff:
+
+- [@Daniel15](https://github.com/Daniel15) has been maintaining [babel-standalone](https://github.com/babel/babel-standalone) which we use in the REPL and has some automation when new releases get out.
+- [@maxiloc](https://github.com/maxiloc) added search functionality via Algolia via [#977](https://github.com/babel/babel.github.io/pull/977)
+
+and added some new collaborators: 
+
+- [@STRML](https://github.com/STRML): Added discourse to all github pages via [#875](https://github.com/babel/babel.github.io/pull/875)
```

Comment:
Nice, I didn't realise this was added 😃 

"Discourse" should have a capital "D" as it's a proper noun.